index.js

```diff
@@ -17,10 +17,15 @@ function createError (code, message, statusCode = 500, Base = Error) {
     if (!new.target) {
       return new FastifyError(...args)
     }
+
     this.code = code
     this.name = 'FastifyError'
     this.statusCode = statusCode
 
+    if (args.length && typeof args.at(-1) === 'object' && 'cause' in args.at(-1)) {
```

Comment:
```suggestion
    const lastElement = args.length - 1
    if (lastElement !== -1 && args[lastElement] && typeof args[lastElement] === 'object' && 'cause' in args[lastElement]) {
```

I added the args[lastElement] for the case that the last element is null.

Reply:
not tested.